api/v1beta1/disruption_types.go

```diff
@@ -433,3 +433,21 @@ func DisruptionIsReinjectable(kind chaostypes.DisruptionKindName) bool {
 
 	return true
 }
+
+// CleanedFreeDisruptions is the list of all disruption kinds where the lifecycle of the failure matches the lifecycle of
+// the chaos pod. So once the chaos pod is gone, there's nothing left for us to clean.
+var CleanedFreeDisruptions = []chaostypes.DisruptionKindName{
+	chaostypes.DisruptionKindNodeFailure,
+	chaostypes.DisruptionKindContainerFailure,
+	chaostypes.DisruptionKindCPUPressure,
+}
+
+func DisruptionIsCleanedForFree(kind string) bool {
+	for _, dis := range CleanedFreeDisruptions {
+		if dis.String() == kind {
+			return true
+		}
+	}
+
+	return false
+}
```

Comment:
There's a trick in go to have such structures where you need to find something in a slice without having to loop over it. You can create a map which has only keys (nil interface will hold no space in memory) so you can look for its key (which is a simple O(1) operation).

```suggestion
var CleanedFreeDisruptions = map[chaostypes.DisruptionKindName]interface{}{
	chaostypes.DisruptionKindNodeFailure: nil,
	chaostypes.DisruptionKindContainerFailure: nil,
	chaostypes.DisruptionKindCPUPressure: nil,
}

func DisruptionIsCleanedForFree(kind string) bool {
    _, found := CleanedFreeDisruptions[chaostypes.DisruptionKindName(kind)]
    
    return found
}
```